### PR TITLE
Prevent order being lost for _nodes API filters

### LINKED ISSUE #41885: Fix cluster API documentation
In the cluster API the following NOTE is wrong (I think):

node filters run in the order in which they are given, which is important if using filters that remove nodes from the set. For example _all,master:false means all the nodes except the master-eligible ones, but master:false,_all means the same as **_all** because the _all filter runs after the master:false filter.

It should be:

node filters run in the order in which they are given, which is important if using filters that remove nodes from the set. For example _all,master:false means all the nodes except the master-eligible ones, but master:false,_all means the same as **master:false** because the _all filter runs after the master:false filter.

### PR DESCRIPTION
**Pull request by astefan**
This PR fixes a bug that made the ordering of `nodeId`s values to be lost. The order in which the node filters are specified is important as it dictates which nodes are being kept and which not in the list for which the metrics are retrieved.

Fixes https://github.com/elastic/elasticsearch/issues/41885.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesInfoAction.java [hunk 1]
```diff
@@ -69,20 +69,28 @@ public String getName() {
...
-            Set<String> metricsOrNodeIds = Strings.tokenizeByCommaToSet(request.param("nodeId", "_all"));
+            String nodeId = request.param("nodeId", "_all");
+            Set<String> metricsOrNodeIds = Strings.tokenizeByCommaToSet(nodeId);
             boolean isMetricsOnly = ALLOWED_METRICS.containsAll(metricsOrNodeIds);
             if (isMetricsOnly) {
                 nodeIds = new String[]{"_all"};
                 metrics = metricsOrNodeIds;
             } else {
-                nodeIds = metricsOrNodeIds.toArray(new String[]{});
+                nodeIds = Strings.splitStringByCommaToArray(nodeId);
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesInfoAction.java]: `splitStringByCommaToArray` does not trim whitespace I think whereas `tokenizeByCommaToSet` did . If we don't want to change the parsing, we should use `tokenizeToStringArray` here I think.

**Review by ywelsch — CHANGES_REQUESTED** [review]
I've left one comment, looking good otherwise

**Comment by astefan** [comment]
@ywelsch thanks. I made the change, but also in a second place where the same parsing method was used.

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**